### PR TITLE
added automatic prefix (based on directory) for definition cache

### PR DESCRIPTION
**Pull request by marcing**

### CODE REVIEW

File: src/Definition/Source/SourceCache.php [hunk 1]
```diff
@@ -23,21 +23,39 @@ class SourceCache implements DefinitionSource, MutableDefinitionSource
      */
     private $cachedSource;
 
+	/**
+	 * @var null|string
+	 */
+	private static $cacheKeyPrefix;
```
Comment (partikus) [inline src/Definition/Source/SourceCache.php]: Is it necessary to use a static property?

File: src/Definition/Source/SourceCache.php [hunk 2]
```diff
@@ -23,21 +23,39 @@ class SourceCache implements DefinitionSource, MutableDefinitionSource
...
+
     public function __construct(DefinitionSource $cachedSource)
     {
         $this->cachedSource = $cachedSource;
     }
 
+	/**
+	 * @return string
+	 */
+    public static function getCacheKey(): string
```
Comment (partikus) [inline src/Definition/Source/SourceCache.php]: if `DefinitionSource` or `MutableDefinitionSource` has no `getCacheKey()` definition I would make it private.

Besides that adding a test case might be useful in the future

File: src/Definition/Source/SourceCache.php (hunk 1 above)
Comment (marcing) [reply src/Definition/Source/SourceCache.php]: yes

File: src/Definition/Source/SourceCache.php (hunk 2 above)
Comment (marcing) [reply src/Definition/Source/SourceCache.php]: it's used outside of the class

Comment (marcing) [reply src/Definition/Source/SourceCache.php]:
```
		if(SourceCache::isSupported())
		{
			$output->write('Clearing apcu...');
			apcu_delete(new APCUIterator('~^' . SourceCache::getCacheKey() . '~'));
			$output->writeln('done.');
		}
```

File: src/Definition/Source/SourceCache.php [hunk 3]
```diff
@@ -23,21 +23,39 @@ class SourceCache implements DefinitionSource, MutableDefinitionSource
...
+     * @return string
+     */
+    public static function getCacheKey(): string
+    {
+        if(self::$cacheKeyPrefix === null)
+        {
+            self::$cacheKeyPrefix = str_replace([':', DIRECTORY_SEPARATOR], '', __DIR__);
+        }
+
+        return self::$cacheKeyPrefix . '_' . self::CACHE_KEY;
```
Comment (falkenhawk) [inline src/Definition/Source/SourceCache.php]: could you list here in a comment an exemplary cache prefix, what's generated from the code, please?

Comment (marcing) [reply src/Definition/Source/SourceCache.php]: homeleadeabkwww.leadersnet.atvendorovosphp-disrcDefinitionSource

Comment (falkenhawk) [reply src/Definition/Source/SourceCache.php]: `homedsdafxxwww.example.comvendorovosphp-disrcDefinitionSource`

**Pull request closed without merging**